src/app/nickname/components/NicknameBox.tsx

```diff
@@ -0,0 +1,72 @@
+import clsx from 'clsx';
+import Image from 'next/image';
+import { FieldErrors, UseFormRegister } from 'react-hook-form';
+
+import { ICON_URL, Input, Text, bgColor } from '@/common';
+
+import { CheckNicknameFormValues } from '../hooks/useCreateNicknameForm';
+
+type NicknameBoxProps = {
+  register: UseFormRegister<CheckNicknameFormValues>;
+  isRequiredText: () => string;
+  hasExclamationMark: (pattern: RegExp) => {
+    value: RegExp;
+    message: string;
+  };
+  isMaxLength: (maxLength: number) => {
+    value: number;
+    message: string;
+  };
+  isMinLength: (minLength: number) => {
+    value: number;
+    message: string;
+  };
+  errors: FieldErrors<CheckNicknameFormValues>;
+  boxDescription: string;
+  defaultValue: string;
+};
+
+export const NicknameBox = ({
+  register,
+  isRequiredText,
+  isMaxLength,
+  isMinLength,
+  hasExclamationMark,
+  errors,
+  boxDescription,
+  defaultValue,
+}: NicknameBoxProps) => {
+  return (
+    <div
+      className={clsx(
+        bgColor['gray110'],
+        'flex w-full flex-col items-center rounded-16px px-16px py-24px'
+      )}
+    >
+      <Image
+        src={ICON_URL.EMOJI_DROOLING}
+        alt="똑스 아이콘"
+        width={52.5}
+        height={52.5}
+      />
+      <div className="mt-16px flex w-full flex-col items-center">
+        <Text typo="headingM" color="white">
+          {boxDescription}
+        </Text>
+        <Input
+          {...register('nickname', {
+            required: isRequiredText(),
+            minLength: isMinLength(2),
+            maxLength: isMaxLength(6),
+            pattern: hasExclamationMark(/^[a-zA-Z0-9가-힣]+$/i),
```

Comment:
```suggestion
          // constants.ts
          export const isEnglishAndKorean = \/^[a-zA-Z0-9가-힣]+$/i\

            pattern: isEnglishAndKorean 
```

음 해당 함수의 역할이 모호한데.. (/^[a-zA-Z0-9가-힣]+$/i) 를 상수로 저장하는건 어떤가요?

Reply:
요거는 에러메세지도 같이 보여줘야해서 value와 message값이 둘다 필요해 input에 바로 넣게되면 코드가 길어질 것 같아 일단 함수로 분리해두겠습니다! 추후에 함수명을 바꾸던가 해볼게용